pkg/sql/execinfrapb/processors_sql.proto

```diff
@@ -255,14 +255,14 @@ message JoinReaderSpec {
   // not maintaining ordering allows for optimizations at lower layers.
   optional bool maintain_ordering = 11 [(gogoproto.nullable) = false];
 
-  // Indicates what implicit system columns this JoinReader is expected to
-  // synthesize. These system columns will be placed at the end of the row
-  // output by the JoinReader, in the order specified.
+  reserved 12;
+  // Indicates whether or not this TableReader is expected to produce any
+  // system columns in its output.
   //
   // This is only used in the special case of index joins, where the final
   // result of the secondary index joined against the primary index is
   // expected to contain the materialized system columns.
-  repeated sqlbase.SystemColumnKind system_columns = 12;
+  optional bool has_system_columns = 13 [(gogoproto.nullable) = false];
```

Comment:
Should bump the distsql version. We're technically not backward compatible because we no longer produce the mvcc column if requested by an old node, so we'll want to bump the min accepted version too.